app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -175,34 +175,68 @@ def update(self, experiment, data):
         return experiment
 
 
-class NimbusStatusRestrictionMixin:
+class NimbusDocumentationLinkSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = NimbusDocumentationLink
+        fields = (
+            "title",
+            "link",
+        )
+
+
+class NimbusExperimentDocumentationLinkMixin:
+    def update(self, experiment, data):
+        documentation_links_data = data.pop("documentation_links", None)
+        experiment = super().update(experiment, data)
+        if documentation_links_data is not None:
+            with transaction.atomic():
+                experiment.documentation_links.all().delete()
+                if documentation_links_data:
+                    for link_data in documentation_links_data:
+                        NimbusDocumentationLink.objects.create(
+                            experiment=experiment, **link_data
+                        )
+
+        return experiment
+
+
+class NimbusStatusValidationMixin:
+    ALLOWS_PUBLISH_STATUS_CHANGE_ONLY = (
+        NimbusExperiment.PublishStatus.REVIEW,
+        NimbusExperiment.PublishStatus.APPROVED,
+        NimbusExperiment.PublishStatus.WAITING,
+    )
     ALLOWS_STATUS_CHANGE_ONLY = (NimbusExperiment.Status.PREVIEW,)
     ALLOWS_UPDATE = (NimbusExperiment.Status.DRAFT,)
-    VALID_STATUS_TRANSITIONS = {
-        NimbusExperiment.Status.DRAFT: (
-            NimbusExperiment.Status.PREVIEW,
-            NimbusExperiment.Status.REVIEW,
-        ),
-        NimbusExperiment.Status.PREVIEW: (
-            NimbusExperiment.Status.DRAFT,
-            NimbusExperiment.Status.REVIEW,
-        ),
-    }
 
     def validate(self, data):
         data = super().validate(data)
+
+        raised_field = None
+        restrictive_statuses = {
+            "status": self.ALLOWS_STATUS_CHANGE_ONLY,
+            "publish_status": self.ALLOWS_PUBLISH_STATUS_CHANGE_ONLY,
+        }
+
         if self.instance:
             status = self.instance.status
-            if status in self.ALLOWS_STATUS_CHANGE_ONLY:
-                if set(data.keys()) != {"status"}:
-                    raise serializers.ValidationError(
-                        {
-                            "experiment": [
-                                f"Nimbus Experiment has status '{status}', only "
-                                "status can be changed."
-                            ]
-                        }
-                    )
+
+            for field, statuses in restrictive_statuses.items():
+                instance_value = getattr(self.instance, field)
+                if instance_value in statuses and set(data.keys()) != {str(field)}:
+                    raised_field = (field, instance_value)
```

Comment:
Ah so this is one of the status fields ya?  I might name `field` as `status_field` I thought as I was reading this it could be any field